featuretools/feature_base/features_deserializer.py

```diff
@@ -73,13 +81,22 @@ def __init__(self, features_dict):
         self._primitives_deserializer = PrimitivesDeserializer()
 
     @classmethod
-    def load(cls, features):
+    def load(cls, features, profile_name):
         if isinstance(features, str):
             try:
                 features_dict = json.loads(features)
             except ValueError:
-                with open(features, 'r') as f:
-                    features_dict = json.load(f)
+                transport_params = {}
+                session = boto3.Session()
+                if isinstance(profile_name, str):
+                    transport_params = {'session': boto3.Session(profile_name=profile_name)}
```

Comment:
when does this code path load the features?

Reply:
at s3.open() or with open() (lines 95 and 98)

Reply:
if `profile_name` is a string then lines 95 and 98 won't get hit because of the if/elif/else structure

Reply:
good catch, will fix!